### PR TITLE
Match InitObjectEventStateFromTemplate

### PR DESCRIPTION
**Pull request by PokeCodec**
One of the needed assignments is redundant, so I put ifndef nonmatching for it.

### CODE REVIEW

File: src/event_object_movement.c [hunk 1]
```diff
@@ -1234,12 +1232,12 @@ static u8 InitObjectEventStateFromTemplate(struct ObjectEventTemplate *template,
     objectEvent->range.as_nybbles.y = template->movementRangeY;
     objectEvent->trainerType = template->trainerType;
     objectEvent->trainerRange_berryTreeId = template->trainerRange_berryTreeId;
+#ifndef NONMATCHING
+    objectEvent->mapNum = mapNum; //Redundant, but needed to match
+#endif
```
Comment (PikalaxALT) [inline src/event_object_movement.c:1237]: No need to wrap this statement in `#ifndef NONMATCHING`. I firmly believe that GameFreak would have done precisely this.

**Comment by PokeCodec** [comment]
Done!

**Comment by PokeCodec** [comment]
@PikalaxALT fixed it. I forgot to pull from the pret repo. git pull fixed the issue

File: src/event_object_movement.c [hunk 2]
```diff
@@ -1205,8 +1205,7 @@ static u8 InitObjectEventStateFromTemplate(struct ObjectEventTemplate *template,
 {
     struct ObjectEvent *objectEvent;
     u8 objectEventId;
-    s16 x;
-    s16 y;
```
Comment (PikalaxALT) [inline src/event_object_movement.c:1209]: Is this change necessary to match?

**Comment by PokeCodec** [comment]
It passed!
